nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ExtractEmailHeaders.java

```diff
@@ -0,0 +1,233 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+
+@EventDriven
+@SideEffectFree
+@Tags({"split", "email"})
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@CapabilityDescription("Using the flowfile content as source of data, extract header from an RFC  compliant  email file adding the relevant attributes to the flowfile. " +
+        "This processor does not perform extensive RFC validation but still requires a bare minimum compliance with RFC 2822")
+@WritesAttributes({
+        @WritesAttribute(attribute = "email.headers.bcc.*", description = "Each individual BCC recipient (if available)"),
+        @WritesAttribute(attribute = "email.headers.cc.*", description = "Each individual CC recipient (if available)"),
+        @WritesAttribute(attribute = "email.headers.from.*", description = "Each individual mailbox contained in the From  of the Email (array as per RFC-2822)"),
+        @WritesAttribute(attribute = "email.headers.message-id", description = "The value of the Message-ID header (if available)"),
+        @WritesAttribute(attribute = "email.headers.received_date", description = "The Received-Date of the message (if available)"),
+        @WritesAttribute(attribute = "email.headers.sent_date", description = "Date the message was sent"),
+        @WritesAttribute(attribute = "email.headers.subject", description = "Subject of the message (if available)"),
+        @WritesAttribute(attribute = "email.headers.to.*", description = "Each individual TO recipient (if available)"),
+        @WritesAttribute(attribute = "email.attachment_count", description = "Number of attachments of the message" )})
+
+public class ExtractEmailHeaders extends AbstractProcessor {
+    public static final String EMAIL_HEADER_BCC = "email.headers.bcc";
+    public static final String EMAIL_HEADER_CC = "email.headers.cc";
+    public static final String EMAIL_HEADER_FROM = "email.headers.from";
+    public static final String EMAIL_HEADER_MESSAGE_ID = "email.headers.message-id";
+    public static final String EMAIL_HEADER_RECV_DATE = "email.headers.received_date";
+    public static final String EMAIL_HEADER_SENT_DATE = "email.headers.sent_date";
+    public static final String EMAIL_HEADER_SUBJECT = "email.headers.subject";
+    public static final String EMAIL_HEADER_TO = "email.headers.to";
+    public static final String EMAIL_ATTACHMENT_COUNT = "email.attachment_count";
+
+    public static final PropertyDescriptor CAPTURED_HEADERS = new PropertyDescriptor.Builder()
+            .name("Header List")
+            .description("COLON separated list of headers to be extracted from the flowfile  content. " +
```

Comment:
This description reads like no headers will be added if this is empty. Should be reworded to point out that these are in addition to the default ones that are always added.